component/site/libraries/dbmodel.php

```diff
@@ -175,7 +175,7 @@ function accessibleCategoryList($aid = null, $catids = null, $catidList = null,
 
 			$dispatcher = JEventDispatcher::getInstance();
 			$dispatcher->trigger('onGetAccessibleCategories', array(& $instances[$index]));
-			if (count($instances[$index]) == 0)
+			if (is_array($instances[$index]) && count($instances[$index]) === 0)
```

Comment:
This should be `if (!is_array($instances[$index]) || count($instances[$index]) === 0)`